src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -49,13 +49,35 @@ def _post(self, request: Request):
             return Response("Relay unauthorized for full config information", 403)
 
         version = request.GET.get("version") or "1"
+
         set_tag("relay_protocol_version", version)
 
-        if self._should_use_v3(version, request):
+        if version == "4":
+            # With v4, we want to move 'measurements' and 'metricsConditionalTagging' into a new
+            # global config.
+            res = self._post_or_schedule_by_key(request)
+            metrics.incr("relay.project_configs.post_v4.pending", amount=len(res["pending"]))
+            metrics.incr("relay.project_configs.post_v4.fetched", amount=len(res["configs"]))
+
+            if request.GET.get("global"):
+                res["global"] = {
+                    "measurements": res["measurements"],
+                    "metricsConditionalTagging": res["metricsConditionalTagging"],
+                }
+
+            del res["measurements"]
+            del res["metricsConditionalTagging"]
+
+            return Response(res, status=200)
+        elif self._should_use_v3(version, request):
             # Always compute the full config. It's invalid to send partial
             # configs to processing relays, and these validate the requests they
             # get with permissions and trim configs down accordingly.
-            return self._post_or_schedule_by_key(request)
```

Comment:
The code for `v3` was easier to read before this change. What do you think if we restore it, and extract common functionality into a separate function instead?

Reply:
I'm a bit confused, the common functionality is already extracted, I removed the two lines specific to 'v3' in '_post_or_schedule_by_key' and reused that function for v4

do you mean encapsulating the extra logic for v3 and v4 into two new functions (and those then calling '_post_or_schedule_by_key' ?

Reply:
Yep, the second case sounds like a good step toward easier readability!

Reply:
ok, i made it so now